src/pip/_internal/utils/appdirs.py

```diff
@@ -18,7 +19,7 @@ def user_cache_dir(appname: str) -> str:
 
 def user_config_dir(appname: str, roaming: bool = True) -> str:
     path = _appdirs.user_config_dir(appname, appauthor=False, roaming=roaming)
-    if _appdirs.system == "darwin" and not os.path.isdir(path):
+    if sys.platform == "darwin" and not os.path.isdir(path):
```

Comment:
`_appdirs.system` handles some edge cases that would break if we switch to `sys.platform`. But I guess it’s not entirely unreasonable to assume people may be OK with that and wait for bug reports if otherwise.

Reply:
https://github.com/ActiveState/appdirs/blob/8eacfa312d77aba28d483fbfb6f6fc54099622be/appdirs.py#L27-L40

Seems to only matter for Jython users, and I think we should bump that conversation over to upstream `platformdirs` to decide on and we can then extract the platform information from there instead of `sys.platform`.

Reply:
Filed https://github.com/platformdirs/platformdirs/issues/43.

Reply:
I removed the `system` attribute from platformdirs when dropping Python 2 and 3.5 support, which also removed jython support as it was only Python 2 at the time  (see https://github.com/platformdirs/platformdirs/pull/14).

Reply:
Oh, then we don't care at all. :)